### DISCUSSION TITLE
Feature request: evaluate multipole expansion on single axis

### DISCUSSION THREAD

**Discussion by nordic-node** [maroba/multipoles#10] [Ideas]
Hi,

instead of evaluating the multipole expansion point by point, like

```python
phi = np.array([mpe(r, 0, 0, l_max=l_max) for r in rr])
``` 

where `mpe` is a multipole expansion object.

I would rather like to write something like this:

```
phi = mpe(rr, 0, 0, l_max=l_max)
```

As far as I can see, the call function just allows to pass a point. and the getitem function wants slices or masks on the same grid
that was used for the charge distribution. Is there a possibility to do the same for outside of that grid?

Cheers, Sara

**Comment by nordic-node** [comment]
I have just sent a pull request implementing this feature. In fact, it implements even more. Now you
can evaluate the multipole expansion not only on an axis but along arbitrary curves. For instance,
evaluating along some segment of the y-axis would be

```
mpe(0, np.linspace(20, 30, 50), 0)
```

But evaluating along a circle of radius 50 in the x-y plane would be

```
t = np.linspace(0, 2*np.pi, 100)
mpe(100*np.cos(t), 100*np.sin(t), 0)
```

**Comment by maroba** [comment]
Yes, I have just merged your pull requests a few minutes ago. Thank you very much! Great work. :-))

**Comment by nordic-node** [comment]
I see that you already have made a new release with my changes. Thanks! By the way, I have a Jupyter notebook showing how the accuracy of your multipole package scales. Shall I post it to the repo as example?

**Comment by maroba** [comment]
That would be awesome!

**Comment by nordic-node** [comment]
Just sent a pull request. 😄
